python/ray/cluster_utils.py

```diff
@@ -100,20 +101,21 @@ def add_node(self, **node_args):
             # We only need one log monitor per physical node.
             ray_params.update_if_absent(include_log_monitor=False)
             # Let grpc pick a port.
-            ray_params.update(node_manager_port=0)
+            ray_params.update_if_absent(node_manager_port=0)
             node = ray.node.Node(
                 ray_params,
                 head=False,
                 shutdown_at_exit=self._shutdown_at_exit,
                 spawn_reaper=self._shutdown_at_exit)
             self.worker_nodes.add(node)
 
-        # Wait for the node to appear in the client table. We do this so that
-        # the nodes appears in the client table in the order that the
-        # corresponding calls to add_node were made. We do this because in the
-        # tests we assume that the driver is connected to the first node that
-        # is added.
-        self._wait_for_node(node)
+        if wait:
```

Comment:
This will slow down CI a lot I guess. Is it to avoid the port conflicting?

Reply:
Ah, if it is for testing, can we just make the default False?

Reply:
(so that we can keep the previous behavior)

Reply:
The previous behavior is to wait always, so I didn't change the default behavior.